### PR TITLE
BN-830_slotData_protobuf_replaceModel

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
Replace  2 specific models, **SlotData, SlotId**

## Approach
Replace old Models for protobuf-specs
## Testing
preparePr
## Tickets
closes BN-830

### CODE REVIEW

File: minting/src/test/scala/co/topl/minting/interpreters/OperationalKeyMakerSpec.scala [hunk 1]
```diff
@@ -49,7 +51,8 @@ class OperationalKeyMakerSpec
       val vrfProof = mock[VrfCalculatorAlgebra[F]]
       val etaCalculation = mock[EtaCalculationAlgebra[F]]
       val consensusState = mock[ConsensusValidationStateAlgebra[F]]
-      val parentSlotId = SlotId(10L, TypedBytes(1: Byte, Bytes.fill(32)(0: Byte)))
+//      val parentSlotId = SlotId(10L, TypedBytes(1: Byte, Bytes.fill(32)(0: Byte)):BlockId) // remove
```
Comment (nandotorterolo) [inline minting/src/test/scala/co/topl/minting/interpreters/OperationalKeyMakerSpec.scala:54]: remove commented line

File: blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala [hunk 2]
```diff
@@ -132,7 +135,7 @@ class BlockchainPeerServerSpec extends CatsEffectSuite with ScalaCheckEffectSuit
         withMock {
           val currentMempoolSet: Set[TypedIdentifier] = ListSet(mempoolTxA, mempoolTxB, mempoolTxC)
           val mempool = mock[MempoolAlgebra[F]]
-          (mempool.read _).expects(head.slotId.blockId).once().returning(currentMempoolSet.pure[F])
+          (mempool.read _).expects(head.slotId.blockId: TypedIdentifier).once().returning(currentMempoolSet.pure[F])
```
Comment (o-aleksandrov-topl) [inline blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala:138]: do we really need all such explicit type `TypedIdentifier`  here and everywhere else?

File: typeclasses/src/main/scala/co/topl/typeclasses/EqInstances.scala [hunk 3]
```diff
@@ -15,6 +18,9 @@ trait EqInstances {
   implicit val bytesEq: Eq[Bytes] =
     (a, b) => a === b
 
+  implicit val bytesStringEq: Eq[ByteString] =
+    (a, b) => ByteString.unsignedLexicographicalComparator().compare(a, b) === 0
+
```
Comment (nandotorterolo) [inline typeclasses/src/main/scala/co/topl/typeclasses/EqInstances.scala:23]: @SeanCheatham what do you think about this Eq implementation?

File: blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala (hunk 2 above)
Comment (nandotorterolo) [reply blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala:138]: I don't like either :( 
We need to find a way to define TypedIdentifier on protobuf, for blockId, HeaderId, TransactionId.
Unfortunately protobuf does not allow to define constants

Comment (nandotorterolo) [reply blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala:138]: Maybe we can include a new field in blockId, with a new rule:

// Ecample for BlockId
// typePrefix must equal 1exactly
> float typePrefix = 1 [(validate.rules).byte.const = 1];
And then in the node, created the Identity type class for each of those

Comment (nandotorterolo) [reply blockchain/src/test/scala/co/topl/blockchain/interpreters/BlockchainPeerServerSpec.scala:138]: long term: remove TypedIdentifier in node

File: typeclasses/src/main/scala/co/topl/typeclasses/EqInstances.scala (hunk 3 above)
Comment (SeanCheatham) [reply typeclasses/src/main/scala/co/topl/typeclasses/EqInstances.scala:23]: I think you can safely use the normal `ByteString#equals` method (`a == b`).  The ByteString's equals method should compare the actual bytes, unlike an Array[Byte].

**Review by ignaciofernandezsoto — APPROVED** [review]

**Pull request merged**